docs/guides/isolated_systems.md

```diff
@@ -0,0 +1,223 @@
+
+3. Deploy to production
+
+    a. Execute the `sqlmesh plan` command to apply the updates to the `prod` environment
+
+![Standard SQLMesh workflow](./isolated_systems/isolated-systems_standard-workflow-staging.png)
+
+When applying the updates to the `prod` environment, all model changes that were previously applied in user environments will not run again due to SQLMesh's virtual data environments.
+
+#### Staging environment
+
+In the workflow described in the previous section, individual code modifications have been tested by the developers that made them, but the collection of all changes being deployed may not have been tested together.
+
+A common expansion of the simple workflow (that we recommend) is to add a staging environment where the set of changes can be evaluated before deploying to production.
+
+The single system workflow with a staging environment is:
+
+1. Developers modify project code
+
+    a. Each developer makes their own copy of the production SQLMesh project files
+
+    b. Developers update code in their copy of the project files
+
+    c. Developers create SQLMesh environments and validate their code updates
+
+2. Update production project code
+
+    a. A set of code updates is selected for deployment to production
+
+    b. The production SQLMesh project files are modified with the updates
+
+    c. Execute the `sqlmesh plan staging` command to apply the updates to the `staging` environment (or another environment of your choosing)
+
+    d. Developers validate the set of code updates in the `staging` environment
+
+3. Deploy to production
+
+    a. Execute the `sqlmesh plan` command to apply the updates to the `prod` environment
+
+![Standard SQLMesh workflow with staging environment](./isolated_systems/isolated-systems_standard-workflow-staging.png)
+
+Because the code updates have run in the `staging` environment, the deployment to the `prod` environment (step 3) will not require additional computation due to SQLMesh's virtual data environments. The deployment only requires updates to views so will be very rapid.
+
+### Workflow with isolated systems
+
+We now describe the SQLMesh project workflow with isolated systems.
+
+The primary difference from the single-system workflows described above is that some steps now occur in the nonproduction system and others in the production system.
+
+In some isolated system implementations, the data in the non-production system is different from that in the production system. For example, columns containing sensitive information may be masked or altered in the non-production system's data. These potential differences increase the likelihood of unintentional errors, so we describe (and recommend) a workflow with staging environments.
+
+In the steps below, we assume that the non-production system is the default gateway. The isolated system workflow with staging environments is:
+
+1. Developers modify project code in the non-production environment
```

Comment:
Wouldnt it be simpler to just call this cloning the git repository. And if not, wouldn't it be simpler to just say that the developer workflow is identical to the prior section barring the fact you should specify the non-prod gateway in the CLI if its not the default. This makes it seem like a very different workflow and the differing verbiage is confusing.

Reply:
Agreed - do the updates make the similarity more apparent?